### PR TITLE
Added index as part of key

### PR DESCRIPTION
**Pull request by wliu080**
This change adds index as a prefix for the 'key' property for most of our maps - this is normally not suggested as it interferes with React's optimizations for list insertions / removals, but in our case we always want to be replacing the whole list since it only occurs on language change.

Note that this is also unlikely to occur to begin with, as it'll only happen when there is a duplicate key (i.e, the same text repeated within an array of text content) which _shouldn't_ occur with the proper copy

### CODE REVIEW

**Comment by wliu080** [comment]
As it's not as high priority, we shouldn't merge this before the featured-topic on landing page PR in case it makes a big merge mess for us

**Review by tombak98 — APPROVED** [review]

**Pull request merged**
